src/components/WeatherApp.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useCallback, useEffect } from "react";
+import styles from "../styles/WeatherApp.module.css";
+import getForecast from "../helpers/getForecast";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import SearchForm from "./SearchForm";
+import ErrorMessage from "./ErrorMessage";
+
+export default function WeatherApp() {
+  const [city, setCity] = useState("");
+  const [location, setLocation] = useState({});
+  const [errorMessage, setErrorMessage] = useState("");
+  const [forecasts, setForecasts] = useState([]);
+  const [selectedDate, setSelectedDate] = useState(0);
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  const acknowledgeError = useCallback(() => {
+    setErrorMessage("");
+    setCity("");
+  }, []);
+
+  const handleForecastSelect = useCallback((date) => {
+    setSelectedDate(date);
+  }, []);
+
+  const updateCity = useCallback((event) => {
+    setCity(event.target.value);
+  }, []);
+
+  const submitCitySearch = useCallback(() => {
+    getForecast({
+      setLocation,
+      setForecasts,
+      setSelectedDate,
+      setErrorMessage,
+      city,
+    });
+  }, [city]);
+
+  useEffect(() => {
+    getForecast({
+      setLocation,
+      setForecasts,
+      setSelectedDate,
+      setErrorMessage,
+      city,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
This can be done a bit differently to avoid the need for the eslint exception. Since this `useEffect` is only being used to populate the initial load of the app, you can provide it with an empty string for the `city` value (since that is what would be passed anyway):
```
  useEffect(() => {
    getForecast({
      setLocation,
      setForecasts,
      setSelectedDate,
      setErrorMessage,
      city: "",
    });
  }, []);
```